main.js

```diff
@@ -40,21 +35,51 @@ define(function (require, exports, module) {
 		});
 	});
 
-	// Eventually, we'll listen for a message from
-	// thimble containing the make's initial code.
-	// For now, we are defaulting to thimble's starter
-	// make.
+	// We wait for thimble to send us the initial make and
+	// write it into the filesystem before allowing brackets
+	// to continue loading. Thimble passes us this information
+	// in the form of: {type: "bramble:init", source: "<CODE HERE>"}
 	exports.initExtension = function() {
 		var deferred = new $.Deferred();
+		var data;
+
+		function _getInitialDocument(e) {
+			try {
+				data = JSON.parse(e.data);
+			} catch(err) {
+				console.error("Parsing message from thimble failed: ", err);
```

Comment:
I wonder if you should set `data = {};` here so that line 58 doesn't blow up when you do `data.type`